subsys/bluetooth/controller/CMakeLists.txt

```diff
@@ -7,6 +7,7 @@
 zephyr_library()
 
 zephyr_library_sources(
+  lock.c
```

Comment:
Rename file to multithreading_lock.c and only include it when CONFIG_MULTITHREADING (not sure what it is called) is selected?